### ISSUE TITLE
client.has_what() not accurate based on UI Status page data

### ISSUE THREAD

**Issue by ameetshah1983** [dask/distributed#1868]
We are trying to throttle the task execution on the grid using the client.has_what() api to determine the free workers.

Seems like the load on workers does not get reflected immediately similar to what is shown on the status UI page( total, in-memory, processing and erred data). At times it takes quite a bit of time for has_what to reflect any data.

Is this the right api to use to find which workers are in use or is there another api that can be used to determine number of free workers which can then be used to determine the throttle range.

**Comment by mrocklin** [comment]
I believe that they both use the same data structures.  

It is uncommon to see people use `Client.has_what` to control execution.  Can I ask more about why you are trying to throttle execution?

**Comment by ameetshah1983** [comment]
We are trying to avoid a single job from utilizing all workers on the grid especially for tasks which submit additional futures for parallelization. For this we were trying to determine the number of free workers at that point in time and parallelize the task to that count and submit additional sub futures on completion of existing ones.
Mainly if we have 15 workers each with 1 core, and a given task has 25 futures to execute in parallel, we want to allocate say only 80% of the free workers to this task so that any new task coming in can be allocated a few workers and not be completely held up by first task.

**Comment by mrocklin** [comment]
If you're able to construct a minimal reproducible example that shows has_what not providing correct values that would be interesting.  I doubt that this is possible though.

Typically in your situation administrators have different users launch their own dask clusters and then use some cluster resource manager to manage their ability to create dask workers.  Dask is rarely used to load balance between workers on its own.

**Comment by ameetshah1983** [comment]
This is a small test i ran.

def test123(test):
    try:
        import time
        i = 0 
        while i <= 50:
            i = i + 1
        time.sleep(i)
        while i <= 100:
            i = i + 1
        return i    
    except:
        return 100

x = [cl.submit(test123, y) for y in range(10)]

On submitting this, my assumption is cl.has_what() should show 10 entries spread across different workers. However, the values dont appear for quite sometime. It seems like they appear after the sleep command is complete to some sort. However the UI status page immediately shows test123 10 executions in progress. 

Please let me know if my understanding of cl.has_what() api is unclear as i am new to dask and still learning its features.

As to your other suggestion of each user managing their own grid is something we are moving towards and set it up similar to what you have mentioned.
